pinot-core/src/test/java/com/linkedin/pinot/core/data/recordtransformer/RecordTransformerTest.java

```diff
@@ -0,0 +1,136 @@
+          .addSingleValueDimension("svString", FieldSpec.DataType.STRING)
+          .addSingleValueDimension("svBytes", FieldSpec.DataType.BYTES)
+          .addMultiValueDimension("mvInt", FieldSpec.DataType.INT)
+          .addMultiValueDimension("mvLong", FieldSpec.DataType.LONG)
+          .addMultiValueDimension("mvFloat", FieldSpec.DataType.FLOAT)
+          .addMultiValueDimension("mvDouble", FieldSpec.DataType.DOUBLE)
+          .addMultiValueDimension("mvString", FieldSpec.DataType.STRING)
+          .addTime("incoming", 6, TimeUnit.HOURS, FieldSpec.DataType.INT, "outgoing", 1, TimeUnit.MILLISECONDS,
+              FieldSpec.DataType.LONG)
+          .build();
+  // Transform multiple times should return the same result
+  private static final int NUM_ROUNDS = 5;
+
+  private static GenericRow getRecord() {
+    GenericRow record = new GenericRow();
+    Map<String, Object> fields = new HashMap<>();
+    fields.put("svInt", (byte) 123);
+    fields.put("svLong", (char) 123);
+    fields.put("svFloat", (short) 123);
+    fields.put("svDouble", "123");
+    fields.put("svString", new Object() {
+      @Override
+      public String toString() {
+        return "123";
+      }
+    });
+    fields.put("svBytes", new byte[]{123});
+    fields.put("mvInt", new Object[]{123L});
+    fields.put("mvLong", new Object[]{123f});
+    fields.put("mvFloat", new Object[]{123d});
+    fields.put("mvDouble", new Object[]{123});
+    fields.put("mvString", true);
+    fields.put("incoming", "123");
+    record.init(fields);
+    return record;
+  }
+
+  @Test
+  public void testDataTypeTransformer() {
+    DataTypeTransformer transformer = new DataTypeTransformer(SCHEMA);
+    GenericRow record = getRecord();
+    for (int i = 0; i < NUM_ROUNDS; i++) {
+      record = transformer.transform(record);
+      assertNotNull(record);
+      assertEquals(record.getValue("svInt"), 123);
+      assertEquals(record.getValue("svLong"), 123L);
+      assertEquals(record.getValue("svFloat"), 123f);
+      assertEquals(record.getValue("svDouble"), 123d);
+      assertEquals(record.getValue("svString"), "123");
+      assertEquals(record.getValue("svBytes"), new ByteArray(new byte[]{123}));
+      assertEquals(record.getValue("mvInt"), new Object[]{123});
+      assertEquals(record.getValue("mvLong"), new Object[]{123L});
+      assertEquals(record.getValue("mvFloat"), new Object[]{123f});
+      assertEquals(record.getValue("mvDouble"), new Object[]{123d});
+      assertEquals(record.getValue("mvString"), new Object[]{"true"});
+    }
+  }
+
+  @Test
+  public void testTimeTransformer() {
+    TimeTransformer transformer = new TimeTransformer(SCHEMA);
+    GenericRow record = getRecord();
+    for (int i = 0; i < NUM_ROUNDS; i++) {
+      record = transformer.transform(record);
+      assertNotNull(record);
+      assertEquals(record.getValue("outgoing"), 123 * 6 * 3600 * 1000L);
+    }
+  }
+
+  @Test
```

Comment:
I am not sure all the removed tests from PlainFieldExtractorTest are covered here. I see some tests there for a random class that has a toString method, and also multiple time-spec transformation tests

Reply:
Here we test the random class with toString() in the "svString" column. Time converter has separate tests (DefaultTimeConverterTest).